src/agni/server/communication/MessageReceiver.java

```diff
@@ -9,20 +17,196 @@
+        this.heartBeatReceiver = heartbeatReceiver;
+        this.channelList = channels;
+    }
+    
+    public enum MessageTypes {
+        HEARTBEAT((byte)0x01),
+        LOGIN((byte)0x02),
+        INFO((byte)0x03),
+        STATUS((byte)0x04),
+        CHAT((byte)0x05),
+        FILE((byte)0x06);
+
+        private final byte bytes;
+        private MessageTypes(byte bytes) {
+            this.bytes = bytes;
+        }
+         final public byte bytes() {
+            return bytes;
+         }
+      }
+
+/*
+ *@requires server port number
+ *@promises selector initialized and ready for connection requests
+ *@promises socket channel is initialized to be non-blocking and bound to specified port
+ */
+    public void initializeConnection(String port) {
+        try {
+            ServerSocketChannel.open();
+            channel.configureBlocking(false);
+            InetSocketAddress isa = new InetSocketAddress (Integer.parseInt(port));
+            channel.socket().bind(isa);
+
+            selector = Selector.open();
+            channel.register(selector, SelectionKey.OP_ACCEPT);
+        } catch (IOException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+    }
+    
+    private void saveIpChannelPair(SocketChannel channel) {
+        String ip = null;
+        try {
+            ip = channel.getRemoteAddress().toString();
+        } catch (IOException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+        channelList.addPair(ip, channel);
+    }
+    
+/*
+ * @requires ip of current channel
+ * @requires message buffer
+ * @promises call appropriate 'receiverMessage(ip, buffer)' method
+ */
+    private void selectReceiver(String ip, ByteBuffer buffer){
+        byte messageType = buffer.get(4);
+        if (messageType == MessageTypes.HEARTBEAT.bytes()) {
+            // Pass to heartbeatReceiver
+            heartBeatReceiver.receiveMessage(ip, buffer);
+        } else if (messageType == MessageTypes.INFO.bytes()) {
+            // Pass to informationReceiver
+            infoRequestReceiver.receiveMessage(ip, buffer);
+        } else if (messageType == MessageTypes.CHAT.bytes()) {
+            // Pass to chatReceiver
+            chatReceiver.receiveMessage(ip, buffer);
+        } else if (messageType == MessageTypes.FILE.bytes()) {
+            // Pass to fileReceiver
+            fileReceiver.receiveMessage(ip, buffer);
+        } else if (messageType == MessageTypes.STATUS.bytes()) {
+            // Pass to statusReceiver
+            userReceiver.receiveMessage(ip, buffer);
+        } else {
+            assert(false);
+        }
+    }
+    
+    /*
+     *accept new client connection request
+     * @requires SelectionKey key
+     * @promises register new client and add new ip/channel pair to list
+     */  
+    public void registerNewClient(SelectionKey key) {
+        SocketChannel cchannel = null;
```

Comment:
I think `cchannel` could use a better name, don't really understand what it's supposed to be.